cleanup/cleanup_test.go

```diff
@@ -0,0 +1,270 @@
+/*
+Copyright 2020 gRPC authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cleanup
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/grpc/test-infra/config"
+)
+
+var _ = Describe("Test Environment", func() {
+	It("supports creation of load tests", func() {
+		err := k8sClient.Create(context.Background(), newLoadTest())
+		Expect(err).ToNot(HaveOccurred())
+	})
+})
+
+var _ = Describe("quitWorkers", func() {
+	var log logr.Logger
+	var podList []*corev1.Pod
+	var driver *corev1.Pod
+	var mockQuit *mockQuitClient
+
+	var clientPending *corev1.Pod
+	var serverPending *corev1.Pod
+	var clientRunning *corev1.Pod
+	var serverRunning *corev1.Pod
+	var clientFailed *corev1.Pod
+	var serverFailed *corev1.Pod
+	var clientUnknown *corev1.Pod
+	var serverUnknown *corev1.Pod
+	var clientSucceeded *corev1.Pod
+	var serverSucceeded *corev1.Pod
+	var ctx context.Context
+
+	BeforeEach(func() {
+		log = ctrl.Log.WithName("CleanupAgent").WithName("LoadTest")
+		mockQuit = &mockQuitClient{}
+		ctx = context.Background()
+
+		podList = []*corev1.Pod{}
```

Comment:
nit: no need to instantiate a podList. You can append to a nil slice in Go. For example,

```go
var colors []string = nil
colors = append(colors, "red")  # a list is lazily initialized when you first append to it here
colors = append(colors, "green")
colors = append(colors, "blue")
```

Reply:
I checked it again, remembered that the reason I initiate the slice every time is trying to have a fresh start for each test.